Index: src/test/resources/features/energy-comparison-test.feature
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/test/resources/features/energy-comparison-test.feature	(date 1575811327135)
+++ src/test/resources/features/energy-comparison-test.feature	(date 1575811327135)
@@ -36,8 +36,8 @@
 #      And the user clicks on the checks more details about the quote
 #      Then all the more details section should be displayed
 
-    Scenario: To verify the user is able to edit the electricity supplier details on the summary page
-      Given the user checks the summary section on the Summary page
-      When the user edits the electricity supplier
-      Then the user lands on the Your Supplier page
-
+#    Scenario: To verify the user is able to edit the electricity supplier details on the summary page
+#      Given the user checks the summary section on the Summary page
+#      When the user edits the electricity supplier
+#      Then the user lands on the Your Supplier page
+#
